### DISCUSSION TITLE
register custom phpstan extensions in rector

### DISCUSSION THREAD

**Discussion by staabm** [rectorphp/rector#6936] [Q&A] [answered]
I have written some phpstan extensions which provide better type coverage with our framework.

is it somehow possible to register these e.g. `DynamicMethodReturnTypeExtension`s within rector, so rector can also make use of this improved type coverage?

**Comment by TomasVotruba** [comment, marked as answer]
Should be via option 'PHPStan config for Rector".

**Reply by staabm** [reply]
I see

https://github.com/rectorphp/rector-src/blob/a84d9511bc69b509039b87fad7f8868c5bef0a93/src/Configuration/Option.php#L147

thx

**Reply by staabm** [reply]
this worked out great thx.

does rector also pickup things like `scanDirectories`  or other directives from these phpstan configs?
or is ist just about defined services?

**Reply by TomasVotruba** [reply]
The goal was for extensions, but it's config like any other, so should work. Can you try?

**Reply by staabm** [reply]
just to confirm, `scanDirectories` etc. works.

I realized my problem was that classes could not be found for the rector to work. it was not about classes which are part of the analyzed code, therefore I needed to provide a proper autoloading via `Option::BOOTSTRAP_FILES`
